# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1297028705 18000
#      Sun Feb 06 16:45:05 2011 -0500
# Node ID 4d30b4136a3e210b5fe6a066640147d15cec0ab3
# Parent  c2e1973d870b25f2ca17d7d6df5ccf4e3eac5058
legend.m: Align legends to plot box, add demo. Bug 32373.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,11 @@
 2011-02-06  Ben Abbott  <bpabbott@mac.com>
 
+	* plot/legend.m: Align legends to plot box, add demo. Bug 32373.
 	* plot/legend.m: "legend off" should delete the legend object.
 
 2011-02-06  David Bateman  <dbateman@free.fr>
 	* plot/legend.m : Delete old legend before probing the position and
 	outerposition. Reshape lpos correct if the legend locqtion is
 	"southeast".
 
 2011-02-06  Ben Abbott  <bpabbott@mac.com>
diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -595,54 +595,53 @@ function [hlegend2, hobjects2, hplot2, t
                             ca_outpos(3) - lpos(3), ca_outpos(4)];
             else
               lpos = [ca_pos(1) +  ypad, ...
                       ca_pos(2) + (ca_pos(4) - lpos(4)) / 2, lpos(3), lpos(4)];
             endif
           case "northeast"
             if (outside)
               lpos = [ca_outpos(1) + ca_outpos(3) - lpos(3), ...
-                      ca_outpos(2) + ca_outpos(4) - lpos(4), lpos(3), lpos(4)];
-              new_pos = [ca_pos(1), ca_pos(2), ca_pos(3) - lpos(3), ...
-                         ca_pos(4) - lpos(4)];
+                      ca_pos(2) + ca_pos(4) - lpos(4), lpos(3), lpos(4)];
+              new_pos = [ca_pos(1), ca_pos(2), ca_pos(3) - lpos(3), ca_pos(4)];
               new_outpos = [ca_outpos(1), ca_outpos(2), ...
-                            ca_outpos(3) - lpos(3), ca_outpos(4) - lpos(4)];
+                            ca_outpos(3) - lpos(3), ca_outpos(4)];
             else
               lpos = [ca_pos(1) + ca_pos(3) - lpos(3) - ypad, ...
                       ca_pos(2) + ca_pos(4) - lpos(4) - ypad, lpos(3), lpos(4)];
             endif
           case "northwest"
             if (outside)
-              lpos = [ca_outpos(1), ca_outpos(2) + ca_outpos(4) - lpos(4), ...
+              lpos = [ca_outpos(1), ca_pos(2) + ca_pos(4) - lpos(4), ...
                       lpos(3), lpos(4)];
               new_pos = [ca_pos(1) + lpos(3), ca_pos(2), ...
-                         ca_pos(3) - lpos(3), ca_pos(4) - lpos(4)];
+                         ca_pos(3) - lpos(3), ca_pos(4)];
               new_outpos = [ca_outpos(1) + lpos(3), ca_outpos(2), ...
-                            ca_outpos(3) - lpos(3), ca_outpos(4) - lpos(4)];
+                            ca_outpos(3) - lpos(3), ca_outpos(4)];
             else
               lpos = [ca_pos(1) + ypad, ...
                       ca_pos(2) + ca_pos(4) - lpos(4) - ypad, lpos(3), lpos(4)];
             endif
           case "southeast"
             if (outside)
-              lpos = [ca_outpos(1) + ca_outpos(3) - lpos(3), ca_outpos(2), ...
+              lpos = [ca_outpos(1) + ca_outpos(3) - lpos(3), ca_pos(2), ...
                       lpos(3), lpos(4)];
-              new_pos = [ca_pos(1), ca_pos(2) + lpos(4), ...
-                         ca_pos(3) - lpos(3), ca_pos(4) - lpos(4)];
+              new_pos = [ca_pos(1), ca_pos(2), ...
+                         ca_pos(3) - lpos(3), ca_pos(4)];
               new_outpos = [ca_outpos(1), ca_outpos(2) + lpos(4), ...
                             ca_outpos(3) - lpos(3), ca_outpos(4) - lpos(4)];
             else
               lpos = [ca_pos(1) + ca_pos(3) - lpos(3) - ypad, ...
                       ca_pos(2) + ypad, lpos(3), lpos(4)];
             endif
           case "southwest"
             if (outside)
-              lpos = [ca_outpos(1), ca_outpos(2), 0, lpos(3), lpos(4)];
-              new_pos = [ca_pos(1) +lpos(3), ca_pos(2) + lpos(4), ...
-                         ca_pos(3) - lpos(3), ca_pos(4) - lpos(4)];
+              lpos = [ca_outpos(1), ca_pos(2), lpos(3), lpos(4)];
+              new_pos = [ca_pos(1) + lpos(3), ca_pos(2), ...
+                         ca_pos(3) - lpos(3), ca_pos(4)];
               new_outpos = [ca_outpos(1) + lpos(3), ca_outpos(2) + lpos(4), ...
                             ca_outpos(3) - lpos(3), ca_outpos(4) - lpos(4)];
             else
               lpos = [ca_pos(1) + ypad, ca_pos(2) + ypad, lpos(3), lpos(4)];
             endif
         endswitch
 
         units = get (hlegend, "units");
@@ -1105,8 +1104,28 @@ endfunction
 %! plot (x, rand (11));
 %! xlabel ("Indices")
 %! ylabel ("Random Values")
 %! title ("Legend ""off"" should delete the legend")
 %! legend (cellstr (num2str ((1:10)')), "location", "northeastoutside")
 %! legend off
 %! axis ([0, 10, 0 1])
 
+%!demo
+%! clf
+%! x = 0:4;
+%! subplot (2, 2, 1)
+%! plot (x, rand (numel (x)));
+%! legend (cellstr (num2str ((1:10)')), "location", "northwestoutside")
+%! legend boxon
+%! subplot (2, 2, 2)
+%! plot (x, rand (numel (x)));
+%! legend (cellstr (num2str ((1:10)')), "location", "northeastoutside")
+%! legend boxon
+%! subplot (2, 2, 3);
+%! plot (x, rand (numel (x)));
+%! legend (cellstr (num2str ((1:10)')), "location", "southwestoutside")
+%! legend boxon
+%! subplot (2, 2, 4)
+%! plot (x, rand (numel (x)));
+%! legend (cellstr (num2str ((1:10)')), "location", "southeastoutside")
+%! legend boxon
+
